### ISSUE TITLE
QWidget::setTabOrder logging to stdout when dialogs are open

### ISSUE THREAD

**Issue by creviera** [freedomofpress/securedrop-client#693]
# STR
1. Click on "Print" or "Export" and see log message: `QWidget::setTabOrder: 'first' and 'second' must be in the same window`

# STR 2

1. Open the client in Offline mode
2. Click on SIGNIN to open the login dialog and see log message: `QWidget::setTabOrder: 'first' and 'second' must be in the same window`

**Comment by ntoll** [comment]
@creviera ack.

**Comment by ntoll** [comment]
OK... I **cannot** reproduce this. Perhaps code in the client has changed in some way. However, I **can** reproduce #697 as reported by @kushal. In any case, I've looked into the cause of this error message and the only reference to it with any meaningful context is from 2009 (https://www.qtcentre.org/threads/18033-QWidget-setTabOrder-question) and an older version of Qt.

I'll see what I can salvage from the old conversation referenced above and check our code in the SourceList/SourceWidget classes. Something about setting the tab layout **after** calling `setLayout`.

**Comment by ntoll** [comment]
TL;DR: is this really worth the rabbit-hole exploration required to remove such warnings..? I suggest we mark as "won't fix" because the benefit will cost lots more investigation.

Further investigation reveals:

* We don't set any tab order related attributes relating to the SourceList or SourceWidget classes.
* We call `setLayout` very early in the life cycle of pretty much all our custom widgets.
* The source of the warning is this line of code in Qt: https://code.woboq.org/qt5/qtbase/src/widgets/kernel/qwidget.cpp.html#7015 Note how it's checking that the widgets share the same `window()`.
* This `window` constant (https://doc.qt.io/qt-5/qwidget.html#window) returns the parent widget in the widget tree that is the "window" of the application.

I'm confused that `window` should be different for two `SourceWidget` widgets in the same `SourceList`.

My advice is to mark this is "won't fix" since it's only emitting to stdout (which the user won't see) and the logic surrounding the `window` feels like a can of worms.

**Comment by redshiftzero** [comment]
fair, I'll remove this ticket from the sprint and mark as low priority

**Comment by eloquence** [comment]
Could there be any relation with #352?

**Comment by creviera** [comment]
I can no longer repro the original STRs but we saw the same behavior when switching between sources in the sourcelist, which I just confirmed https://github.com/freedomofpress/securedrop-client/pull/1261 fixes. Since https://github.com/freedomofpress/securedrop-client/pull/1261 has been merged, I'm closing this issue.

**Comment by gonzalo-bulnes** [comment]
I hadn't seen this issue! Yes, it does look like #1261 a lot. I understood well the ordering requirement @creviera, so if helpful I'm happy to walk you or someone else through it if it shows up again 🙂
